### PR TITLE
[cumulative] AB#33546 - Closed Grid Gaps

### PR DESCRIPTION
**Pull request by morethanfire**
In this PR
- Grid
    - Converted Grid SCSS to JSS
    - Added breakpoints to our Grid system.
    - Added material-ui’s spacing method with our own custom size. 
    - Updated Grid documentation.
    - Fixed areas in our components and docs where Grid is being used.
- DataGrid
    - Converted DataGrid’s SCSS to JSS
    - Added DataGrid documentation
- Card
    - Converted Card’s SCSS to JSS
- ActionBar
    - Converted ActionBar’s SCSS to JSS
    - Added breakpoints to ActionBar props 
    - Added ActionBar documentation

### CODE REVIEW

**Comment by groberts314** [comment]
PR Review Checkpoint Meeting Thurs 2020-08-20.
Geoffrey to review.

File: src/surfaces/actionBar/actionBar.jsx [hunk 1]
```diff
@@ -36,28 +53,410 @@ const propTypes = {
...
+    justifyContent: PropTypes.oneOf([
+        'center',
+        'flex-end',
+        'flex-start',
+        'space-around',
+        'space-between',
+        'space-evenly',
+    ]),
     moduleType: PropTypes.oneOf(['drawer', 'page']),
     style: PropTypes.shape({}), // eslint-disable-line react/forbid-prop-types
```
Comment (groberts314) [inline src/surfaces/actionBar/actionBar.jsx:75]: Is `// eslint-disable-line react/forbid-prop-types` still required here if we're using `shape({})`?  It _shouldn't_ be ... right?

File: src/surfaces/actionBar/actionBar.jsx [hunk 2]
```diff
@@ -36,28 +53,410 @@ const propTypes = {
...
+    toggleSmSearchVisbileClassName: PropTypes.func,
```
Comment (groberts314) [inline src/surfaces/actionBar/actionBar.jsx:76]: "Visible" is misspelled in this prop name.  Should be `toggleSmSearchVisibleClassName`.

File: docs/package.json [hunk 3]
```diff
@@ -13,6 +13,7 @@
     "classnames": "^2.2.6",
     "core-js": "^3.1.3",
     "lodash": "^4.17.19",
+    "moment-timezone": "^0.5.21",
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply docs/package.json:16]: Fixed.

File: docs/src/dataDisplay/icon/icon.jsx [hunk 4]
```diff
@@ -311,7 +315,14 @@ class ElementsIconSet extends React.PureComponent {
...
 
     render() {
-        const { classes } = this.props;
+        const {
+            classes,
+            width,
+        } = this.props;
+
+        console.log('width', width);
+        console.log('window.innerWidth', window.innerWidth);
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply docs/src/dataDisplay/icon/icon.jsx]: Fixed.

File: docs/src/dataDisplay/icon/icon.jsx [hunk 5]
```diff
@@ -329,32 +340,66 @@ class ElementsIconSet extends React.PureComponent {
...
-                <Grid.Column key={index} laptop={3} mobileLarge={4} width={6}>
-                    <Icon compact={iconCompact} size={iconSize} type={iconType} />
+                <React.Fragment
+                    key={index}
+                >
+                    {width === 'lg' && index % 5 === 0 ? (
+                        <Grid.Column
+                            lg={12}
+                        />
+                    ) : null}
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply docs/src/dataDisplay/icon/icon.jsx:349]: Yeah, for now we are only wanting to go to five columns at the lg breakpoint. The md breakpoint is a little to small for five column. Ideally a custom breakpoint between md and lg for this doc would be good, but for now I think we are good.

File: docs/src/global/example.jsx [hunk 6]
```diff
@@ -12,7 +12,11 @@ import Highlighter from './highlighter';
 
 const propTypes = {
     children: PropTypes.node.isRequired,
-    rawCode: PropTypes.string.isRequired,
+    rawCode: PropTypes.string,
+};
+
+const defaultProps = {
+    rawCode: null,
 };
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply docs/src/global/example.jsx]: Yes, for the interactive examples.

File: docs/src/global/componentVersionIdentifier.jsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+const propTypes = {
+    pathname: PropTypes.string.isRequired,
+};
+
+function ComponentVersionIdentifier(props) {
+    const {
+        pathname,
+    } = props;
+
+    const pthNmArray = filter(pathname.split('/'), (segment) => segment);
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply docs/src/global/componentVersionIdentifier.jsx]: fixed.

File: docs/src/layout/grid/grid.jsx [hunk 8]
```diff
@@ -1,1301 +1,177 @@
...
+import { default as docGridColumn } from '!!@advclb/react-docgen-loader!react-cm-ui/layout/grid/gridColumn';
+/* eslint-enable import/no-named-default, import/extensions */
+
+const propTypes = {
+    location: PropTypes.shape({
+        pathname: PropTypes.string,
+    }).isRequired,
+};
+
+function DocsTable(props) {
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply docs/src/layout/grid/grid.jsx:34]: Fixed.

File: src/dataDisplay/dataGrid/dataGrid.jsx [hunk 9]
```diff
@@ -57,75 +65,121 @@ const defaultProps = {
...
+        classNameModuleModifier,
+        className,
+        {
+            [classes.drawer]: moduleType === 'drawer',
+            [classes.bleed]: bleed,
+            [classes.stickyColumns]: isStickyColumns,
+        },
+    );
+
+    console.log('sortable', sortable);
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply src/dataDisplay/dataGrid/dataGrid.jsx]: Fixed.

File: src/layout/grid/grid.jsx [hunk 10]
```diff
@@ -1,85 +1,250 @@
...
+        },
+        ...gutters,
+    };
+});
+
+/**
+ * The Grid layout responsivly adapts to screen size, aiding in dividing up content into their own
+ * regions.
+ */
 const Grid = React.forwardRef((props, ref) => {
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply src/layout/grid/grid.jsx]: Fixed.

File: src/surfaces/actionBar/actionBar.jsx (hunk 2 above)
Comment (morethanfire) [reply src/surfaces/actionBar/actionBar.jsx:76]: fixed.

File: src/surfaces/actionBar/actionBar.jsx (hunk 1 above)
Comment (morethanfire) [reply src/surfaces/actionBar/actionBar.jsx:75]: Fixed.

File: docs/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply docs/package.json]: Looks good.  Resolving.

File: docs/src/dataDisplay/icon/icon.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply docs/src/dataDisplay/icon/icon.jsx]: Looks good.  Resolving.

File: docs/src/global/componentVersionIdentifier.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply docs/src/global/componentVersionIdentifier.jsx]: Yeah, I think I saw this fixed in the other branches.  Resolving.

File: docs/src/layout/grid/grid.jsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply docs/src/layout/grid/grid.jsx]: Looks good.  Resolving.

File: src/dataDisplay/dataGrid/dataGrid.jsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply src/dataDisplay/dataGrid/dataGrid.jsx]: I don't see any more diffs in this file, so this must be resolved.

File: src/layout/grid/grid.jsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply src/layout/grid/grid.jsx]: Looks good.  Resolving.

File: src/layout/grid/gridColumn.jsx [hunk 11]
```diff
@@ -1,113 +1,168 @@
...
-    mobile: PropTypes.oneOf(columnEnums),
-    mobileLarge: PropTypes.oneOf(columnEnums),
-    mobileMedium: PropTypes.oneOf(columnEnums),
-    style: PropTypes.shape({}),
-    tablet: PropTypes.oneOf(columnEnums),
-    textAlign: PropTypes.oneOf(textAlignEnums),
-    verticalAlign: PropTypes.oneOf(verticalAlignEnums),
-    width: PropTypes.oneOf(columnEnums),
+    /**
+     * Defines the number of grids the component is going to use.
```
[the remark this answers is not in the document]
Comment (groberts314) [reply src/layout/grid/gridColumn.jsx]: Looks good.  Resolving.

File: src/surfaces/actionBar/actionBar.jsx (hunk 1 above)
Comment (groberts314) [reply src/surfaces/actionBar/actionBar.jsx]: Looks good.  Resolving.

File: src/surfaces/actionBar/actionBar.jsx (hunk 2 above)
Comment (groberts314) [reply src/surfaces/actionBar/actionBar.jsx]: Looks good.  Resolving.

**Comment by groberts314** [comment]
I think we've arrived at :shipit: .
Nice work!  Take _that_ UI Gaps! 😁

**Pull request merged**